feat(#1797): add max width property to components

bdfranck opened the pull request:
This PR adds a max width property to the following components:

- Accordion
- Callout
- Checkbox
- Container
- Details
- Form item
- Radio item

#### Questions

1. I don't see a test file for a React radio item, just a [radio group](https://github.com/GovAlta/ui-components/blob/alpha/libs/react-components/src/lib/radio-group/radio-group.spec.tsx). Is there a reason for this? Should I add tests to the radio group?
2. I didn't include the dropdown component because I'm not sure where and how to add the max width. Any suggestions? I was thinking we could add it to a separate PR.

#### PR checklist
- [x] I have run a build locally.
- [x] I have ran unit tests locally.
![image](https://github.com/GovAlta/ui-components/assets/1479091/c161e089-5766-48fc-8fea-e07c164a15c3)

- [x] I have tested the functionality.

bdfranck commented:
I've double checked all the files for correct casing and format.

### Accordion
Web component
- [ ] Name is lower case
- [ ] CSS is valid

React component
- [ ] WC name is lower case
- [ ] React name is camel case
- [ ] Prop declaration is proper case

### Callout
Web component
- [ ] Name is lower case
- [ ] CSS is valid

React component
- [ ] WC name is lower case
- [ ] React name is camel case
- [ ] Prop declaration is proper case

### Checkbox
Web component
- [ ] Name is lower case
- [ ] CSS is valid

React component
- [ ] WC name is lower case
- [ ] React name is camel case
- [ ] Prop declaration is proper case

### Container
Web component
- [ ] Name is lower case
- [ ] CSS is valid

React component
- [ ] WC name is lower case
- [ ] React name is camel case
- [ ] Prop declaration is proper case

### Details
Web component
- [ ] Name is lower case
- [ ] CSS is valid

React component
- [ ] WC name is lower case
- [ ] React name is camel case
- [ ] Prop declaration is proper case

### Form item
Web component
- [ ] Name is lower case
- [ ] CSS is valid

React component
- [ ] WC name is lower case
- [ ] React name is camel case
- [ ] Prop declaration is proper case

### Radio item
Web component
- [ ] Name is lower case
- [ ] CSS is valid

React component
- [ ] WC name is lower case
- [ ] React name is camel case
- [ ] Prop declaration is proper case

### Text area
Web component
- [ ] Name is lower case
- [ ] CSS is valid

React component
- [ ] WC name is lower case
- [ ] React name is camel case
- [ ] Prop declaration is proper case

bdfranck commented:
Testing functionality...

### Accordion
I can set a max width on the web component
I can set a max width on the react compontent.

### Callout
I can set a max width on the web component
I can set a max width on the react compontent.

### Checkbox
I can set a max width on the web component
I can set a max width on the react compontent.

### Container
I can set a max width on the web component
I can set a max width on the react compontent.

### Details
I can set a max width on the web component
I can set a max width on the react compontent.

### Form item
I can set a max width on the web component
I can set a max width on the react compontent.

### Radio item
I can set a max width on the web component
I can set a max width on the react compontent.

### Text area
I can set a max width on the web component
I can set a max width on the react compontent.

lizhuomeng71 commented:
When Radio Group at orientation="horizontal", set labelMaxWidth or descriptionMaxWidth at percentage, it will leave a blank space after the element

Code
```
      <GoARadioGroup name="item" value="1" onChange={onChange} orientation="horizontal">
        <GoARadioItem
          value="1"
          label="Wraps an input control with a text label, requirement label, helper text, and error text. Wraps an input control with a text label, requirement label, helper text, and error text. Wraps an input control with a text label, requirement label, helper text, and error text. Wraps an input control with a text label, requirement label, helper text, and error text."
          labelMaxWidth="50%"
          descriptionMaxWidth="50%"
          description="Wraps an input control with a text label, requirement label, helper text, and error text. Wraps an input control with a text label, requirement label, helper text, and error text. Wraps an input control with a text label, requirement label, helper text, and error text. Wraps an input control with a text label, requirement label, helper text, and error text."
        >
        </GoARadioItem>
        <GoARadioItem value="2" label="Label 3"></GoARadioItem>

      </GoARadioGroup>
```
![image](https://github.com/GovAlta/ui-components/assets/7528736/64804769-4912-45b1-81a0-9328e435222e)

bdfranck commented:
I've updated my PR and removed the label, help text, and description max widths. The checkbox, radio item, and form item now have a single max width property.

bdfranck commented:
Re-testing the checkbox, radio item, and form item...

I can set a max width on the checkbox web component. 👍 
<img width="547" alt="image" src="https://github.com/GovAlta/ui-components/assets/1479091/42844afc-ac75-4453-9b6c-91f448edb86f">

I can set a max width on the checkbox react component. 👍 
<img width="515" alt="image" src="https://github.com/GovAlta/ui-components/assets/1479091/f8380098-4ef8-4f28-a894-72095e2ecdad">

I can set a max width on the radio web component. 👍 
<img width="815" alt="image" src="https://github.com/GovAlta/ui-components/assets/1479091/819a815a-31e9-4aff-a14e-de3a4e2b8feb">

I can set a max width on the radio react component. 👍 
<img width="514" alt="image" src="https://github.com/GovAlta/ui-components/assets/1479091/1e9eb939-8148-46f3-a8d3-bd2ca83159c0">

I can set a max width on the form item web component. 👍 
<img width="511" alt="image" src="https://github.com/GovAlta/ui-components/assets/1479091/816a6a62-36cf-48ed-8edf-5bc695746c35">

I can set a max width on the form item react component. 👍 
<img width="522" alt="image" src="https://github.com/GovAlta/ui-components/assets/1479091/b98fbb21-4f16-4d82-8bc9-af38c44b5ad2">

lizhuomeng71 commented:
When GoARadioGroup has orientation="horizontal"

the maxWidth will create an extra spacing

![image](https://github.com/GovAlta/ui-components/assets/7528736/d801588a-80c0-4198-abe4-bfbf77b4af13)

'''
    <GoAFormItem label="Basic">
      <GoARadioGroup name="item" value="1" onChange={onChange} orientation="horizontal">
        <GoARadioItem
          value="1"
          maxWidth="30%"
          label="Wraps an input control with a text label, requirement label, helper text, and error text. Wraps an input control with a text label, requirement label, helper text, and error text. Wraps an input control with a text label, requirement label, helper text, and error text. Wraps an input control with a text label, requirement label, helper text, and error text."
          labelMaxWidth="20%"
          descriptionMaxWidth="100%"
          description="Wraps an input control with a text label, requirement label, helper text, and error text. Wraps an input control with a text label, requirement label, helper text, and error text. Wraps an input control with a text label, requirement label, helper text, and error text. Wraps an input control with a text label, requirement label, helper text, and error text."
        >
        </GoARadioItem>
        <GoARadioItem value="2" label="Label 2"></GoARadioItem>
        <GoARadioItem value="3" label="Label 3"></GoARadioItem>
      </GoARadioGroup>
    </GoAFormItem>
'''

bdfranck commented:
I've resolved the merge conflicts on my branch.

bdfranck commented:
@lizhuomeng71 Hmmm, that issue only seems to occur when you enter a percentage value. Here's how it looks when I enter a `px` value:

<img width="723" alt="image" src="https://github.com/user-attachments/assets/63817004-7962-4a5e-b31d-5c4447dfb545">

I'll see if I can spot an issue or if this just a general issue with with CSS flexbox.

bdfranck commented:
@lizhuomeng71 I spent some time exploring the issue with a relative width. The issue is that it's calculating the percentage against its immediate parent, `goa-radio-item`, not `goa-radio-group`.

This is working as expected since the [percentage](https://developer.mozilla.org/en-US/docs/Web/CSS/width) "Defines the width as a percentage of the [containing block](https://developer.mozilla.org/en-US/docs/Web/CSS/Containing_block)'s width."

I would advise teams against using `%` units in this situation.

![image](https://github.com/user-attachments/assets/4bf52bd5-cdb7-4ac5-b7cd-74e2b9dab61d)

lizhuomeng71 commented:
The issue has been verified with the July 11th commit for both React and Angular. It has also been tested and confirmed on the desktop and mobile versions (using Chrome emulator).

bdfranck commented:
I've resolved the merge conflict on this PR.

bdfranck commented:
@vanessatran-ddi I've updated the PR by adding `maxwidth` documentation to the relevant HTML custom data files.

bdfranck commented:
I've updated my PR by removing an `{id}` property that I accidentally re-added in `formItem.svelte` when resolving a merge conflict.

bdfranck commented:
I also removed the `id` test from `formItem.spec.ts`. 👍 The tests all pass again.

tzuge commented:
:tada: This PR is included in version 1.17.0-alpha.88 :tada:

The release is available on:
- [npm package (@alpha dist-tag)](https://www.npmjs.com/package/@abgov/web-components/v/1.17.0-alpha.88)
- [GitHub release](https://github.com/GovAlta/ui-components/releases/tag/web-components-v1.17.0-alpha.88)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

tzuge commented:
:tada: This PR is included in version 4.17.0-alpha.34 :tada:

The release is available on:
- [npm package (@alpha dist-tag)](https://www.npmjs.com/package/@abgov/react-components/v/4.17.0-alpha.34)
- [GitHub release](https://github.com/GovAlta/ui-components/releases/tag/react-v4.17.0-alpha.34)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

tzuge commented:
:tada: This PR is included in version 1.24.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@abgov/web-components/v/1.24.0)
- [GitHub release](https://github.com/GovAlta/ui-components/releases/tag/web-components-v1.24.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

tzuge commented:
:tada: This PR is included in version 4.23.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@abgov/react-components/v/4.23.0)
- [GitHub release](https://github.com/GovAlta/ui-components/releases/tag/react-v4.23.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
